src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -386,8 +386,14 @@ private async Task ProcessItemSelectionAsync(NavigationBarItem item, Cancellatio
                     item.Spans = item.TrackingSpans.SelectAsArray(ts => ts.GetSpan(snapshot).Span.ToTextSpan());
                     var view = _presenter.TryGetCurrentView();
 
-                    // ConfigureAwait(true) as we have to come back to UI thread in order to kick of the refresh task below.
-                    await navBarService.NavigateToItemAsync(document, item, view, cancellationToken).ConfigureAwait(true);
+                    // ConfigureAwait(true) as we have to come back to UI thread in order to kick of the refresh task
+                    // below. Note that we only want to refresh if selecting the item had an effect (either navigating
+                    // or generating).  If nothing happened to don't want to refresh.  This is important as some items
+                    // exist in the type list that are only there to show a set a particular set of items in the member
+                    // list.  So selecting such an item should only update the member list, and we do not want a refresh
+                    // to wipe that out.
+                    if (!await navBarService.TryNavigateToItemAsync(document, item, view, cancellationToken).ConfigureAwait(true))
+                        return;
```

Comment:
this regressed in:  https://github.com/dotnet/roslyn/pull/52530/files#diff-902dbb57c78c6bbd71f7409a96a3db1dbd4813fb5658a069ef3b3b33da063a8aL305

THe reason for this is that the old code effectively would say: ok, something was selected.  However the semantic version of the solution is the same, so i won't refresh anything.

The approach here is cleaner.  If trying to navigate failed, there's no need for us to have our controller change anything, and we'll instead just let the view update accordingly.